File: python/patterns/mvc.py
class CounterModel:
    def __init__(self) -> None:
        self.value = 0

    def increment(self) -> None:
        self.value += 1


class CounterView:
    def render(self, value: int) -> None:
        print(f"Counter value: {value}")


class CounterController:
    def __init__(self, model: CounterModel, view: CounterView) -> None:
        self.model = model
        self.view = view

    def increment(self) -> None:
        self.model.increment()
        self.view.render(self.model.value)


def run() -> None:
    print("\nModel View Controller")
    model = CounterModel()
    view = CounterView()
    controller = CounterController(model, view)
    controller.increment()
    controller.increment()

File: csharp/Patterns/Mvc.cs
using System;

namespace DesignPatterns.Patterns;

public static class Mvc
{
    private class CounterModel
    {
        public int Value { get; private set; }

        public void Increment() => Value++;
    }

    private class CounterView
    {
        public void Render(int value) => Console.WriteLine($"Counter value: {value}");
    }

    private class CounterController
    {
        private readonly CounterModel _model;
        private readonly CounterView _view;

        public CounterController(CounterModel model, CounterView view)
        {
            _model = model;
            _view = view;
        }

        public void Increment()
        {
            _model.Increment();
            _view.Render(_model.Value);
        }
    }

    public static void Run()
    {
        Console.WriteLine("\nModel View Controller");
        var model = new CounterModel();
        var view = new CounterView();
        var controller = new CounterController(model, view);
        controller.Increment();
        controller.Increment();
    }
}
